examples/src/main/java/com/mapbox/navigation/examples/core/BaseRouterActivityJava.java

```diff
@@ -0,0 +1,273 @@
+
+  @BindView(R.id.mapView)
+  MapView mapView;
+
+  abstract Router setupRouter();
+
+  @Override
+  protected void onCreate(@Nullable Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_mock_navigation);
+    ButterKnife.bind(this);
+
+    router = setupRouter();
+
+    mapView.onCreate(savedInstanceState);
+    mapView.getMapAsync(this);
+  }
+
+  @OnClick(R.id.newLocationFab)
+  public void onNewLocationClick() {
+    newOrigin();
+  }
+
+  private void newOrigin() {
+    if (mapboxMap != null) {
+      clearMap();
+      LatLng latLng = Utils.getRandomLatLng(new double[] { -77.1825, 38.7825, -76.9790, 39.0157 });
+      origin = Point.fromLngLat(latLng.getLongitude(), latLng.getLatitude());
+      mapboxMap.moveCamera(CameraUpdateFactory.newLatLngZoom(latLng, 12));
+    }
+  }
+
+  @Override
+  public void onMapReady(@NonNull MapboxMap mapboxMap) {
+    this.mapboxMap = mapboxMap;
+    this.mapboxMap.addOnMapClickListener(this);
+    mapboxMap.setStyle(Style.MAPBOX_STREETS, style -> {
+      Drawable image = ContextCompat.getDrawable(this, R.drawable.mapbox_marker_icon_default);
+      style.addImage(MARKER_ROUTE, image);
+      navigationMapRoute = new NavigationMapRoute(mapView, mapboxMap);
+      symbolManager = new SymbolManager(mapView, mapboxMap, style);
+      Snackbar.make(findViewById(R.id.container), "Tap map to place waypoint", Snackbar.LENGTH_LONG)
+          .show();
+      newOrigin();
+    });
+  }
+
+  private void clearMap() {
+    symbolManager.deleteAll();
+    route = null;
+    destination = null;
+    waypoint = null;
+    navigationMapRoute.updateRouteVisibilityTo(false);
+    navigationMapRoute.updateRouteArrowVisibilityTo(false);
+  }
+
+  private void findRoute() {
+    if (origin != null && destination != null) {
+      router.cancel();
+      if (TurfMeasurement.distance(origin, destination, TurfConstants.UNIT_METERS) > 50) {
+        RouteOptions.Builder optionsBuilder = applyDefaultParams(RouteOptions.builder())
+            .accessToken(Utils.getMapboxAccessToken(this))
+            .annotations(
+                DirectionsCriteria.ANNOTATION_CONGESTION + ","
+                    + DirectionsCriteria.ANNOTATION_DISTANCE
+                    + "," + DirectionsCriteria.ANNOTATION_DURATION
+            );
+
+        List<Point> waypoints = new ArrayList<>();
+        waypoints.add(waypoint);
+
+        coordinates(optionsBuilder, origin, waypoints, destination);
+
+        router.getRoute(optionsBuilder.build(), new Router.Callback() {
+          @Override
+          public void onResponse(@NotNull List<? extends DirectionsRoute> routes) {
+            if (!routes.isEmpty()) {
+              navigationMapRoute.addRoute(routes.get(0));
+            }
+          }
+
+          @Override
+          public void onFailure(@NotNull Throwable throwable) {
+            Timber.e(throwable, "onRoutesRequestFailure: navigation.getRoute()");
+          }
```

Comment:
We missed the `Toast` and `MapboxLogger` 👀 https://github.com/mapbox/mapbox-navigation-android/blob/422e0a05bb65f16f7fd78bbeaf6087f2f40fa57a/examples/src/main/java/com/mapbox/navigation/examples/core/OffboardRouterActivityJava.java#L170-L174

Reply:
OnboardActivity didn't has such logic, should I add it to base one?

Reply:
Yeah, why not? It's mainly showcasing how to use `MapboxLogger` 👍

Reply:
fixed